commit 54519b1eea4e9c40bb2f0cb2ff7c0f841bedf379
Author: Dave Hansen <dave.hansen@linux.intel.com>
Date:   Thu Feb 23 14:26:03 2017 -0800

    x86/pkeys: Check against max pkey to avoid overflows
    
    commit 58ab9a088ddac4efe823471275859d64f735577e upstream.
    
    Kirill reported a warning from UBSAN about undefined behavior when using
    protection keys.  He is running on hardware that actually has support for
    it, which is not widely available.
    
    The warning triggers because of very large shifts of integers when doing a
    pkey_free() of a large, invalid value. This happens because we never check
    that the pkey "fits" into the mm_pkey_allocation_map().
    
    I do not believe there is any danger here of anything bad happening
    other than some aliasing issues where somebody could do:
    
            pkey_free(35);
    
    and the kernel would effectively execute:
    
            pkey_free(8);
    
    While this might be confusing to an app that was doing something stupid, it
    has to do something stupid and the effects are limited to the app shooting
    itself in the foot.
    
    Signed-off-by: Dave Hansen <dave.hansen@linux.intel.com>
    Cc: linux-kselftest@vger.kernel.org
    Cc: shuah@kernel.org
    Cc: kirill.shutemov@linux.intel.com
    Link: http://lkml.kernel.org/r/20170223222603.A022ED65@viggo.jf.intel.com
    Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1297_54519b1eea4e9c40bb2f0cb2ff7c0f841bedf379.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1297_54519b1eea4e9c40bb2f0cb2ff7c0f841bedf379.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
